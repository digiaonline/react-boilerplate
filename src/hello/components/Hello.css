@import '../../styles/colors.css';
@import '../../styles/typography.css';

.component {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: var(--silver);
}

.title {
  font-family: var(--font-serif);
  font-size: rem(36px);
}
