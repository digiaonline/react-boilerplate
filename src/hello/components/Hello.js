// @flow

import flow from 'lodash/flow'
import React, {Component} from 'react'
import {connect} from 'react-redux'
import {sayHello} from '../actions'
import {helloSelector} from '../selectors'
import css from './Hello.css'

type HelloProps = {
  message: string,
  sayHello: typeof sayHello,
}

export class Hello extends Component {
  props: HelloProps

  static defaultProps = {
    message: 'Hello from React.',
  }

  componentWillMount() {
    this.props.sayHello()
  }

  render() {
    const {message} = this.props

    return (
      <div className={css.component}>
        <h1 className={css.title}>{message}</h1>
      </div>
    )
  }
}

export default flow(connect(helloSelector, {sayHello}))(Hello)
