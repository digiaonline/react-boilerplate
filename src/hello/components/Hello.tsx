import * as React from 'react'
import { observer } from 'mobx-react'
import helloStore from '../store'
import * as css from './Hello.css'

type Props = {}
type State = {}

@observer
export class Hello extends React.Component<Props, State> {
  componentWillMount() {
    helloStore.sayHello()
  }

  render() {
    return (
      <div className={css.component}>
        <h1 className={css.title}>{helloStore.message}</h1>
      </div>
    )
  }
}

export default Hello
