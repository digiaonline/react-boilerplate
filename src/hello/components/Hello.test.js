import React from 'react'
import renderer from 'react-test-renderer'
import {Hello} from './Hello'

test('Hello component should read message from MobX store', () => {
  const tree = renderer.create(<Hello />).toJSON()
  expect(tree).toMatchSnapshot()
})
