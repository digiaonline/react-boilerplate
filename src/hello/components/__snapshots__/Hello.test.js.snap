// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello component should set message from props 1`] = `
<div
  className={undefined}
>
  <h1
    className={undefined}
  >
    Hello from Jest.
  </h1>
</div>
`;
